function_app.py

```diff
@@ -153,6 +153,9 @@ def forward(azeventhub: func.EventHubEvent):
     except json.decoder.JSONDecodeError as _:
         logging.info('EVENT=%s', data)
         data = {'records' : [{'Category': 'unsupported', 'Message': data}]}
+    if type(data) == list:

```

Comment:
should we do something like 
```suggestion
    if type(data) not in [<supported_types>] :

```

Reply:
`list` is actually the only case remaining. We deserialise the payload assuming it's json before this check. Then we assume later that it is a json object and validate that relevant keys are part of it. This change is to cover the case where deserialising as json succeeds but the obtained object is a list rather than a json object.